### PR TITLE
Revise the code for training with "torch.nn.BCELoss()"

### PR DESCRIPTION
**Pull request by minseok-ryu**
Note that "torch.nn.BCELoss()" requires "targets" to be float numbers and reshaped as follows:

                if self.loss_type == "torch.nn.BCELoss()":
                    target = target.to(torch.float32)                                
                    loss = self.loss_fn(output, target.reshape(-1,1))
                else:
                    loss = self.loss_fn(output, target)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -172,3 +174,18 @@ examples/gcloud/generated_pb2/federated_learning_pb2_grpc.py
...
+examples/gcloud/Dockerfile
+examples/gcloud/Dockerfile.repo
+examples/gcloud/Dockerfile.repo1
+examples/gcloud/Dockerfile.repo
+examples/gcloud/Dockerfile
+examples/gcloud/Dockerfile
+examples/gcloud/Dockerfile.covid
+examples/gcloud/Dockerfile.mnist
+examples/gcloud/Dockerfile.repo
+examples/gcloud/Dockerfile
```
Comment (kibaekkim) [inline .gitignore:191]: can you clean up these?

File: examples/deepcovid.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+    def forward(self, x):
+        x = self.act(self.conv1(x))
+        x = self.maxpool(x)
+        x = self.act(self.conv2(x))
+        x = self.maxpool(x)
+        x = torch.flatten(x, 1)
+        x = self.act(self.fc1(x))
+        x = self.fc2(x)
+        return x
```
Comment (kibaekkim) [inline examples/deepcovid.py:53]: Shall we remove this?

File: examples/deepcovid.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        
+    # train_datasets = []
+    # train_datasets.append(Dataset( torch.FloatTensor(train_data_raw["x"]), torch.tensor(train_data_raw["y"]) ))
+        
+    # # test data 
+    # with open("./datasets/PreprocessedData/deepcovid_test_data.json") as f:
+    #     test_data_raw = json.load(f)
+     
+    # test_dataset = Dataset( torch.FloatTensor(test_data_raw["x"]), torch.tensor(test_data_raw["y"]) )
+
```
Comment (kibaekkim) [inline examples/deepcovid.py:88]: clean up?

File: examples/deepcovid.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+from models.cnn import *
+
+import appfl.run as rt
+from mpi4py import MPI
+
+DataSet_name = "DeepCovid"
+num_clients = 1
+num_channel = 3  # 1 if gray, 3 if color
+num_classes = 2  # number of the image classes
+num_pixel = 32  # image size = (num_pixel, num_pixel)
```
Comment (kibaekkim) [inline examples/deepcovid.py:22]: These are not used once we remove CNN.

File: examples/gcloud/covid_binary_client_1.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    # model = CNN(3, 2, 32)
+    # cfg.fed.args.loss_type = "torch.nn.CrossEntropyLoss()"
+     
+    # with open("../datasets/PreprocessedData/deepcovid_train_data.json") as f:
+    #     train_data_raw = json.load(f)
+    
+    # train_dataset = Dataset(
+    #         torch.FloatTensor(train_data_raw["x"]), torch.tensor(train_data_raw["y"])
+    #     )
+
```
Comment (kibaekkim) [inline examples/gcloud/covid_binary_client_1.py:81]: clean up?

File: examples/gcloud/covid_binary_client_2.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    # """ CNN """    
+    # model = CNN(3, 2, 32)
+    # cfg.fed.args.loss_type = "torch.nn.CrossEntropyLoss()"
+     
+    # with open("../datasets/PreprocessedData/deepcovid_test_data.json") as f:
+    #     train_data_raw = json.load(f)
+    
+    # train_dataset = Dataset(
+    #         torch.FloatTensor(train_data_raw["x"]), torch.tensor(train_data_raw["y"])
+    #     )
```
Comment (kibaekkim) [inline examples/gcloud/covid_binary_client_2.py:78]: clean up

File: examples/gcloud/covid_binary_server.py [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+    def forward(self, x):
+        x = self.act(self.conv1(x))
+        x = self.maxpool(x)
+        x = self.act(self.conv2(x))
+        x = self.maxpool(x)
+        x = torch.flatten(x, 1)
+        x = self.act(self.fc1(x))
+        x = self.fc2(x)
+        return x
```
Comment (kibaekkim) [inline examples/gcloud/covid_binary_server.py:46]: shall we remove this?

File: examples/gcloud/covid_binary_server.py [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    cfg.num_epochs = args.total_iter
+    cfg.fed.args.num_local_epochs = args.local_iter
+    cfg.fed.args.optim_args.lr = args.lr
+
+    logging.basicConfig(stream=sys.stdout, level=eval("logging." + args.logging))
+    
+    start_time = time.time()    
+
+    # """ CNN """    
+    # model = CNN(3, 2, 32)
```
Comment (kibaekkim) [inline examples/gcloud/covid_binary_server.py:73]: clean up

File: examples/gcloud/grpc_mnist_server.py [hunk 9]
```diff
@@ -49,6 +49,7 @@ def forward(self, x):
 def main():
 
     parser = argparse.ArgumentParser(description="Provide the configuration")
+    parser.add_argument("--total_iter", type=int, required=True)
```
Comment (kibaekkim) [inline examples/gcloud/grpc_mnist_server.py:52]: why don't we use the same name (`num_epochs`) as defined in `cfg`?

File: examples/gcloud/covid_binary_client_2.py [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+    logger.debug(OmegaConf.to_yaml(cfg))
+
+    grpc_client.run_client(
+        cfg, args.client_id, model, train_dataset
+    )
+    logger.info("------DONE------")
+
+
+if __name__ == "__main__":
+    main()
```
Comment (kibaekkim) [inline examples/gcloud/covid_binary_client_2.py:116]: What is the difference between this and `covid_binary_client_1.py`? A brief description to each file may be helpful.

File: src/appfl/algorithm/fedavg.py [hunk 11]
```diff
@@ -69,8 +69,8 @@ def update(self):
                 target = target.to(self.device)
 
                 optimizer.zero_grad()
-                output = self.model(data)
-                loss = self.loss_fn(output, target)
+                output = self.model(data)                
+                loss = self.loss_fn(output, target)
```
Comment (kibaekkim) [inline src/appfl/algorithm/fedavg.py:73]: unnecessary white spaces... can you run black formatting? maybe for all the files that are updated?

File: .gitignore (hunk 1 above)
Comment (minseok-ryu) [reply .gitignore:191]: Done

File: examples/deepcovid.py (hunk 2 above)
Comment (minseok-ryu) [reply examples/deepcovid.py:53]: done

File: examples/deepcovid.py (hunk 3 above)
Comment (minseok-ryu) [reply examples/deepcovid.py:88]: done

File: examples/deepcovid.py (hunk 4 above)
Comment (minseok-ryu) [reply examples/deepcovid.py:22]: done

File: examples/gcloud/covid_binary_client_1.py [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+    def forward(self, x):
+        x = self.act(self.conv1(x))
+        x = self.maxpool(x)
+        x = self.act(self.conv2(x))
+        x = self.maxpool(x)
+        x = torch.flatten(x, 1)
+        x = self.act(self.fc1(x))
+        x = self.fc2(x)
+        return x
```
[the remark this answers is not in the document]
Comment (minseok-ryu) [reply examples/gcloud/covid_binary_client_1.py:47]: done

File: examples/gcloud/covid_binary_client_1.py (hunk 5 above)
Comment (minseok-ryu) [reply examples/gcloud/covid_binary_client_1.py:81]: done

File: examples/gcloud/covid_binary_client_2.py [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+    def forward(self, x):
+        x = self.act(self.conv1(x))
+        x = self.maxpool(x)
+        x = self.act(self.conv2(x))
+        x = self.maxpool(x)
+        x = torch.flatten(x, 1)
+        x = self.act(self.fc1(x))
+        x = self.fc2(x)
+        return x
+
```
[the remark this answers is not in the document]
Comment (minseok-ryu) [reply examples/gcloud/covid_binary_client_2.py:48]: removed this file

File: examples/gcloud/covid_binary_server.py (hunk 7 above)
Comment (minseok-ryu) [reply examples/gcloud/covid_binary_server.py:46]: done

File: examples/gcloud/covid_binary_server.py (hunk 8 above)
Comment (minseok-ryu) [reply examples/gcloud/covid_binary_server.py:73]: done

File: examples/gcloud/grpc_mnist_server.py (hunk 9 above)
Comment (minseok-ryu) [reply examples/gcloud/grpc_mnist_server.py:52]: done

File: examples/gcloud/covid_binary_client_2.py (hunk 10 above)
Comment (minseok-ryu) [reply examples/gcloud/covid_binary_client_2.py:116]: It just had different dataset. The file is removed.

**Review by kibaekkim — COMMENTED** [review]
Let's address these:
- remove all covid examples
- configure loss function in the mnist example with some description
